src/query/plan/operator.cpp

```diff
@@ -6063,7 +6095,50 @@ bool PeriodicSubquery::PeriodicSubqueryCursor::Pull(Frame &frame, ExecutionConte
   OOMExceptionEnabler oom_exception;
   SCOPED_PROFILE_OP("PeriodicSubquery");
 
-  throw utils::NotYetImplemented("periodic commit");
+  if (commit_frequency_ == -1) {
+    ExpressionEvaluator evaluator(&frame, context.symbol_table, context.evaluation_context, context.db_accessor,
```

Comment:
What do you think about moving this commit_frequence calculation to a common function which can be reused for `PeriodicCommitCursor` and `PeriodicSubqueryCursor`?